src/sagas/pushNotification.js

```diff
@@ -444,6 +445,28 @@ export function* dismissOptInQuestion() {
   yield STORE.setItem(pushNotificationKey.optInDismissed, true);
 }
 
+const cleanToken = async () => {
+  await messaging().unregisterDeviceForRemoteMessages();
+  await messaging().deleteToken();
+};
+
+/**
+ * This function is responsible for reset the push notification.
+ */
+export function* resetPushNotification() {
+  // Unregister the device from FCM
+  yield call(cleanToken);
+  // Clean the store
+  yield STORE.removeItem(pushNotificationKey.enabledAt);
+  yield STORE.removeItem(pushNotificationKey.settings);
+  yield STORE.removeItem(pushNotificationKey.hasBeenEnabled);
+  yield STORE.removeItem(pushNotificationKey.optInDismissed);
```

Comment:
Why are you removing this? This should continue to be saved because you need to show the modal only the first time the user opens his wallet with the new feature, not every new initialized wallet.

Reply:
We should not reset the push notification feature after a wallet reset?

Reply:
This is required especially because we reset the device ID as well.

Reply:
Isn't the device ID the same if the user resets the wallet? The idea for the opt in modal is like a message to the user that we have a new feature available. It shouldn't appear every time the user loads a wallet, otherwise it's not good in my opinion.

Reply:
Solved it by removing the optInDismissed from reset.